egendata/app
# feat: 🎸 use react-native-jose

- use [native implementation of methods for JOSE](https://github.com/egendata/react-native-jose)

- [x] `sign`
- [x] `verify`
- [x] `decode`
- [x] `remove jwt-lite`


How to test locally:

- Android: 
    - `npm i`
    - `npm start`
    - `npm run android` (atleast needed first time since it runs Jetifier to migrate libraries to AndroidX; after that you can run it from Android Studio)
    - if you make changes inside the react-native-jose, sometimes, due to some caching I assume, the app crashes on start (so go to `MainApplication.java` file and comment out `new JosePackage()` and build the app, then uncomment it back and build again and it should clear the cache and build it)
    - if you want to run it on an actual device you need to run `adb reverse tcp:8081 tcp:8081` so that the phone can reach the Metro bundler

- iOS:
    - `npm i`
    - `cd ios && pod install && cd ..`
    - open the workspace in XCode and then you can choose to run it on a simulator or directly on an iPhone
